src/java/main/org/apache/zookeeper/server/NettyServerCnxn.java

```diff
@@ -1,856 +0,0 @@
-    {
-        // We take advantage of the limited size of the length to look
-        // for cmds. They are all 4-bytes which fits inside of an int
-        String cmd = cmd2String.get(len);
-        if (cmd == null) {
-            return false;
-        }
-        channel.setInterestOps(0).awaitUninterruptibly();
-        LOG.info("Processing " + cmd + " command from "
-                + channel.getRemoteAddress());
-        packetReceived();
-
-        final PrintWriter pwriter = new PrintWriter(
-                new BufferedWriter(new SendBufferWriter()));
-        if (len == ruokCmd) {
-            RuokCommand ruok = new RuokCommand(pwriter);
-            ruok.start();
-            return true;
-        } else if (len == getTraceMaskCmd) {
-            TraceMaskCommand tmask = new TraceMaskCommand(pwriter);
-            tmask.start();
-            return true;
-        } else if (len == setTraceMaskCmd) {
-            ByteBuffer mask = ByteBuffer.allocate(4);
-            message.readBytes(mask);
-
-            bb.flip();
-            long traceMask = mask.getLong();
-            ZooTrace.setTextTraceLevel(traceMask);
-            SetTraceMaskCommand setMask = new SetTraceMaskCommand(pwriter, traceMask);
-            setMask.start();
-            return true;
-        } else if (len == enviCmd) {
-            EnvCommand env = new EnvCommand(pwriter);
-            env.start();
-            return true;
-        } else if (len == confCmd) {
-            ConfCommand ccmd = new ConfCommand(pwriter);
-            ccmd.start();
-            return true;
-        } else if (len == srstCmd) {
-            StatResetCommand strst = new StatResetCommand(pwriter);
-            strst.start();
-            return true;
-        } else if (len == crstCmd) {
-            CnxnStatResetCommand crst = new CnxnStatResetCommand(pwriter);
-            crst.start();
-            return true;
-        } else if (len == dumpCmd) {
-            DumpCommand dump = new DumpCommand(pwriter);
-            dump.start();
-            return true;
-        } else if (len == statCmd || len == srvrCmd) {
-            StatCommand stat = new StatCommand(pwriter, len);
-            stat.start();
-            return true;
-        } else if (len == consCmd) {
-            ConsCommand cons = new ConsCommand(pwriter);
-            cons.start();
-            return true;
-        } else if (len == wchpCmd || len == wchcCmd || len == wchsCmd) {
-            WatchCommand wcmd = new WatchCommand(pwriter, len);
-            wcmd.start();
-            return true;
-        } else if (len == mntrCmd) {
-            MonitorCommand mntr = new MonitorCommand(pwriter);
-            mntr.start();
-            return true;
-        } else if (len == isroCmd) {
-            IsroCommand isro = new IsroCommand(pwriter);
-            isro.start();
-            return true;
-        }
-        return false;
-    }
-
-    public void receiveMessage(ChannelBuffer message) {
```

Comment:
### ✅ Getting better: Bumpy Road Ahead
receiveMessage decreases from 6 to 5 logical blocks with deeply nested code, threshold is one single block per function